lib/pos_mempool.go

```diff
@@ -174,11 +173,11 @@ func (dmp *PosMempool) IsRunning() bool {
 // AddTransaction validates a MsgDeSoTxn transaction and adds it to the mempool if it is valid.
 // If the mempool overflows as a result of adding the transaction, the mempool is pruned.
 func (dmp *PosMempool) AddTransaction(txn *MsgDeSoTxn) error {
-	if !dmp.IsRunning() {
-		return errors.Wrapf(MempoolErrorNotRunning, "PosMempool.AddTransaction: ")
-	}
+	// First, validate that the transaction is properly formatted according to BalanceModel. We acquire a read lock on
+	// the mempool. This allows multiple goroutines to safely perform transaction validation concurrently. In particular,
+	// transaction signature verification can be parallelized.
+	dmp.RLock()
 
-	// First, validate that the transaction is properly formatted according to BalanceModel.
 	if err := ValidateDeSoTxnSanityBalanceModel(txn, dmp.latestBlockHeight, dmp.params, dmp.globalParams); err != nil {
 		return errors.Wrapf(err, "PosMempool.AddTransaction: Problem validating transaction sanity")
```

Comment:
What if we return early here? Then `dmp.RUnlock()` below won't get called so the read lock will be held indefinitely

Reply:
Damn, good catch! Fixed in #660